src/TLuaInterpreter.cpp

```diff
@@ -1777,6 +1777,8 @@ int TLuaInterpreter::deleteLine(lua_State* L)
 int TLuaInterpreter::saveMap(lua_State* L)
 {
     string location = "";
+    int saveVersion = 0;
+
     if (lua_gettop(L) == 1) {
```

Comment:
If you test against the number of options with a `> 0` here you only need to test that the first argument (___if supplied___) __is__ a string once in the code (and you won't need to ensure that you have the same error message in both places) and, like Vadi suggests, - it should be something like:

    lua_pushfstring(L, "saveMap: bad argument #1 type (save location path and file name as string is optional, got %s!)", luaL_typename(L, 1));